src/errors/BaseError.ts

```diff
@@ -1,14 +1,20 @@
 class BaseIsomerError extends Error {
```

Comment:
shouldn't this `implement IsomerInternalError`

Reply:
hmm wait quick question, do we not want to benefit from inheritance here? why are we preferring `implement` over `extends` ah

Reply:
was planning to add in future pr. added now

Reply:
@kishore03109 - we `extends` a `class` but `implement` an `interface` (which is what we have here)